nncf/torch/model_transformer.py

```diff
@@ -0,0 +1,97 @@
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.graph.transformations.commands import TransformationCommand
+from nncf.common.graph.transformations.commands import TransformationPriority
+from nncf.torch.graph.transformations.commands import PTBiasCorrectionCommand
+from nncf.torch.graph.transformations.commands import PTInsertionCommand
+from nncf.torch.graph.transformations.commands import PTModelExtractionCommand
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.graph.transformations.layout import PTTransformationLayout
+from nncf.torch.module_operations import UpdateWeight
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.nncf_network import PTInsertionPoint
+
+
+class PTModelTransformer(ModelTransformer):
+    def __init__(self, model: NNCFNetwork):
+        super().__init__(model)
+        self._node_to_op_address_mapping = model.get_node_to_op_address_mapping()
+
+    def transform(self, transformation_layout: PTTransformationLayout) -> NNCFNetwork:
+        transformations = transformation_layout.transformations
+        insertion_transformations = []
+        extraction_transformations = []
+        bias_correction_transformations = []
+        for transformation in transformations:
+            if isinstance(transformation, PTInsertionCommand):
+                insertion_transformations.append(transformation)
+            if isinstance(transformation, PTModelExtractionCommand):
+                extraction_transformations.append(transformation)
+            if isinstance(transformation, PTBiasCorrectionCommand):
+                bias_correction_transformations.append(transformation)
+
+        if extraction_transformations:
+            return self._apply_extraction_transformations(extraction_transformations)
+
+        if insertion_transformations:
+            self._apply_output_insertion_transformations(insertion_transformations)
+        if bias_correction_transformations:
+            self._apply_bias_correction_transformations(bias_correction_transformations)
+        return self._model
+
+    def _apply_output_insertion_transformations(self, transformations: List[TransformationCommand]) -> None:
+        fns_grouped_by_points = {}  # type: Dict[PTInsertionPoint, List[Tuple[Callable, TransformationPriority]]]
+        for transformation_command in transformations:  # type: PTInsertionCommand
+            target_point = transformation_command.target_point  # type: PTTargetPoint
+            target_node_name = target_point.target_node_name
+            pt_ip = PTInsertionPoint(
+                target_type=target_point.target_type,
+                op_address=self._node_to_op_address_mapping[target_node_name],
+                input_port_id=target_point.input_port_id,
+            )
+            fn = transformation_command.fn
+            if target_point.type is TargetType.OPERATION_WITH_WEIGHTS:
+                fn = UpdateWeight(fn)
```

Comment:
The method is called `_apply_output_insertion_transformations`. What is `UpdateWeight` doing here?

Reply:
UpdateWeight is operator that contains FQ module.

Reply:
`output_insertion` inside method name isn't reflecting meaning of this method. It inserts hooks rather that outputs

Reply:
And it's code that just moved from nncf_network.py and moved to function.

Reply:
What means the new name `_apply_insertion_transformations`? Is it inserts `outputs`? Or, maybe `quantizers`?
Also, how is the `UpdateWeight` connected with the `insertion_transformations`?

Reply:
`_apply_insertion_transformations` means that it works for all `PTInsertionCommand`. Is can insert all that support PTInsertionCommand.  

> Also, how is the UpdateWeight connected with the insertion_transformations?

Just `UpdateWeight` used for insertion target point to pytorch model in some cases.

https://github.com/openvinotoolkit/nncf/blob/818ad9c29e6c31e0a1c8cf33df8529dd92454e0b/nncf/torch/nncf_network.py#L950-L975

Reply:
Since all OV and ONNX backends have different types of commands for output- and quantizer insertions, I've expected to see the same behavior in PyTorch. 

> Just UpdateWeight used for insertion target point to pytorch model in some cases.

That means that the name of the method is not correct. Let's fix it. Also, target points are needed not only for `weight update`.

Reply:
> Since all OV and ONNX backends have different types of commands for output- and quantizer insertions, I've expected to see the same behavior in PyTorch.

Where in FBC used output insertion? Quantizer? How it connected with this PR?

> That means that the name of the method is not correct. 
Why? Could you explain me what wrong in naming function as `_apply_insertion_transformations` for `PTInsertionCommand`? And how `UpdateWeight` is connected with it?

> Also, target points are needed not only for weight update.
So `target points` used not only for `weight update`.

I dont see no any reason to rework using `PTInsertionCommand` in this PR.

Reply:
> How are the _apply_insertion_transformations connected with the UpdateWeight?

Adding `UpdateWeight` operator to nn.Module is not insertion transformation? Yes, why not? 

May be the problem that you think that `UpdateWeight` is something like `BiasCorrection` that once change value of weight. 
But no, it's used for adding FQ or PruningMasks modules to the PT node, that calls on each forward.
It's way in PT to transform the model. It's used because PT models is not a graph. 
 
https://github.com/openvinotoolkit/nncf/blob/fbfd7ed7b3c1192b08011bbc5e8a96cc59897804/nncf/torch/module_operations.py#L61-L67
 
is it confusing your?

I really don't understand what problem here? 
Please, explain clearly what is wrong and what you propose.

Reply:
For me, the `UpdateWeight` class/method/etc., naming is wrong, because it's not clear and straightforward in terms of functionality. I mean, it's not obvious that `UpdateWeight` does something related to the `Quantizer`. The naming of such variables is not transparent in terms of usage in `insertion transformation`.

> It's used because PT models is not a graph.

It's not a problem. Just name such methods as `AddQuantizer` or `AddPruningMask`, or whatever.

Reply:
Ok, It's only your option about name of core module that exists before NNCF was published on Github. 

How i connected to `UpdateWeight` and why i should change it now?

Reply:
I prefer to stop this discussion now. If you don't care about the code understanding, it's up to you.